### PR TITLE
matchers: add range matchers

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

File: xds/type/matcher/v3/range.proto [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// semantics [start, end).
+message Int64Range {
+  // start of the range (inclusive)
+  int64 start = 1;
+
+  // end of the range (exclusive)
+  int64 end = 2;
+}
+
+// Specifies a set of ranges for matching an int64 number.
```
Comment (markdroth) [inline xds/type/matcher/v3/range.proto:24]: Are these sublinear matchers or linear matchers?  If sublinear, what are the precedence rules when there are multiple matching ranges?  Please document.

**Review by markdroth — COMMENTED** [review]
Thanks for doing this!

File: xds/type/matcher/v3/range.proto [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+option java_package = "com.github.xds.type.matcher.v3";
+option java_outer_classname = "RangeProto";
+option java_multiple_files = true;
+option go_package = "github.com/cncf/xds/go/xds/type/matcher/v3";
+
+// [#protodoc-title: Range matcher]
+
+// Specifies the int64 start and end of the range using half-open interval
+// semantics [start, end).
+message Int64Range {
```
[the remark this answers is not in the document]
Comment (kyessenov) [reply xds/type/matcher/v3/range.proto]: Done.

File: xds/type/matcher/v3/range.proto (hunk 1 above)
Comment (kyessenov) [reply xds/type/matcher/v3/range.proto]: First one wins. Thanks. Yes, this is intended to be implemented with a segment tree or the likes.

**Comment by kyessenov** [comment]
Gentle ping.

**Review by markdroth — APPROVED** [review]
Sorry for the delay.  Looks great!

**Pull request merged**